pentagon/component/__init__.py

```diff
@@ -0,0 +1,53 @@
+import os
+import jinja2
+import glob
+import shutil
+import logging
+import traceback
+import sys
+from pentagon.helpers import render_template
+
+
+class ComponentBase():
+    """ Base class for Pentagon Components. """
+    _required_parameters = []
+
+    def __init__(self, data, additional_args=None, **kwargs):
+
+        self._data = data
+        self._additional_args = additional_args
+
+        missing_parameters = []
+        for item in self._required_parameters:
+            if item not in self._data.keys():
+                missing_parameters.append(item)
+
+        if missing_parameters:
+            logging.error("Missing required data parameters: {}".format(",".join(missing_parameters)))
+            sys.exit(1)
+
+    @property
+    def _destination_directory_name(self):
+        if self._destination != './':
+            return self._destination
+        return self._data.get('name', self.__class__.__name__.replace('.', '_').lower())
```

Comment:
What is the case where `__name__` can have `.` in it?

Reply:
Its something I ran into before I was using `self.__class__.__name__`, when I was using `component_path` instead. The replace got left behind unnecessarily.